data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -74,10 +70,13 @@ static String post(CloseableHttpClient httpClient, String urlStr, String payload
             if (entity != null) {
                 StatusLine statusLine = response.getStatusLine();
                 String responseContent = EntityUtils.toString(entity);
-                if (statusLine.getStatusCode() == 200) {
-                    return responseContent;
-                } else {
-                    throw createExceptionFromResponse(urlStr, response, null, responseContent);
+                switch (statusLine.getStatusCode()) {
+                    case HttpStatus.SC_OK:
+                        return responseContent;
+                    case HttpStatus.SC_TOO_MANY_REQUESTS:
```

Comment:
just make sure we really send this form dm

Reply:
also - make sure that the last exception (after 4 attempts ) will be thrown to the user as a DataServiceException with the http status code and all as if it was thrown before this change